fabric-input-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/input/client/InputUtilMixin.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.event.input.client;
+
+import java.util.Map;
+
+import org.lwjgl.glfw.GLFW;
+import org.lwjgl.glfw.GLFWCharModsCallbackI;
+import org.lwjgl.glfw.GLFWKeyCallbackI;
+import org.lwjgl.glfw.GLFWCursorPosCallbackI;
+import org.lwjgl.glfw.GLFWDropCallback;
+import org.lwjgl.glfw.GLFWMouseButtonCallbackI;
+import org.lwjgl.glfw.GLFWScrollCallbackI;
+import org.lwjgl.glfw.GLFWDropCallbackI;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+
+import net.fabricmc.fabric.api.event.client.input.CharEvent;
+import net.fabricmc.fabric.api.event.client.input.ClientInputEvents;
+import net.fabricmc.fabric.api.event.client.input.KeyEvent;
+import net.fabricmc.fabric.api.event.client.input.KeybindEvent;
+import net.fabricmc.fabric.api.event.client.input.MouseButtonEvent;
+import net.fabricmc.fabric.api.event.client.input.MouseMoveEvent;
+import net.fabricmc.fabric.api.event.client.input.MouseScrollEvent;
+import net.fabricmc.fabric.impl.client.input.FabricKeyboardImpl;
+import net.fabricmc.fabric.impl.client.input.FabricMouseImpl;
+
+@Mixin(InputUtil.class)
+public class InputUtilMixin {
+	@ModifyVariable(method = "setKeyboardCallbacks(JLorg/lwjgl/glfw/GLFWKeyCallbackI;Lorg/lwjgl/glfw/GLFWCharModsCallbackI;)V", at = @At("HEAD"), index = 2)
+	private static GLFWKeyCallbackI changeKeyCb(GLFWKeyCallbackI keyCb) {
+		return (window, code, scancode, action, mods) -> {
```

Comment:
Please inject into synthetic/lambda methods of `Keyboard` instead.

Same for the mouse ones and `Mouse`.

Reply:
Fair enough

Reply:
Yeah, this is way more destructive than Fabric wants to be.